### ISSUE TITLE
hexrd command line gives suspect completeness values

### ISSUE THREAD

**Issue by donald-e-boyce** [HEXRD/hexrd#506]
After running fit-grains (cli) and comparing with the GUI grains.out file, the command line version gives lower completenesses.  The data was the multiruby data set in the hexrd-examples repository, and the command was:
```
fit-grains -f mruby_config_monolithic.yml
```
Here is the grains.out file from the GUI. The completnesses are 0.989, 1.0 and 1.0.  
```
# grain ID    completeness  chi^2         exp_map_c[0]             exp_map_c[1]             exp_map_c[2]             t_vec_c[0]               t_vec_c[1]               t_vec_c[2]               inv(V_s)[0,0]            inv(V_s)[1,1]            inv(V_s)[2,2]            inv(V_s)[1,2]*sqrt(2)    inv(V_s)[0,2]*sqrt(2)    inv(V_s)[0,2]*sqrt(2)    ln(V_s)[0,0]             ln(V_s)[1,1]             ln(V_s)[2,2]             ln(V_s)[1,2]             ln(V_s)[0,2]             ln(V_s)[0,1]           
0             0.989583      6.933991e-04  1.2148387137022127e+00   6.9419335365732548e-01   7.6558824380266421e-01   6.1698189346522228e-02   1.1750468412117730e-01   -6.9448498961853539e-02  1.0003346870351610e+00   9.9953319615665381e-01   9.9997736626284484e-01   3.1860836343896659e-04   -3.6641496245884612e-05  1.8650848604597950e-04   -3.3462200874956452e-04  4.6694692453028603e-04   2.2659715890254105e-05   -2.2534699809700172e-04  2.5920266729365987e-05   -1.3189305664404040e-04
1             1.000000      6.150296e-04  1.4103808957806463e+00   -1.4274680888986194e-01  -3.7294317834466262e-01  2.0722802063347281e-02   -1.2736923835414615e-01  3.4717666301938938e-01   1.0003276974599085e+00   9.9968897427058556e-01   1.0001420872798406e+00   8.3210432200100929e-05   -1.2012300278129713e-04  6.2691865845444563e-05   -3.2763919086542238e-04  3.1107682240434333e-04   -1.4207184931256043e-04  -5.8845515089365661e-05  8.4921147437906443e-05   -4.4331974241242868e-05
2             1.000000      7.665692e-04  3.4406264579077889e-01   8.1660124882132956e-01   -1.6815972520098272e-02  1.3321311348764454e-02   -1.1878903779374898e-01  -3.3498387922503675e-01  9.9975835120868051e-01   1.0001080766206727e+00   1.0002231756831392e+00   4.2326344384117427e-04   -5.5317220210240794e-04  7.0469477422153352e-04   2.4187872492314113e-04   -1.0790181622566737e-04  -2.2302948336976603e-04  -2.9934038681428545e-04  3.9123006270730111e-04   -4.9838635242633938e-04
```

Here is the grains.out from the command line. The completeness are near 80%, which seems way wrong. And the goodness of fit values (chi-squared) are two or three times larger.

However, the orientations are very close, as are the centroids, and the command line strains actually seem a little better than the GUI (since instrument resolution is approximately 1e-4, and so the log strain values should be near or under that).

```
# grain ID    completeness  chi^2         exp_map_c[0]             exp_map_c[1]             exp_map_c[2]             t_vec_c[0]               t_vec_c[1]               t_vec_c[2]               inv(V_s)[0,0]            inv(V_s)[1,1]            inv(V_s)[2,2]            inv(V_s)[1,2]*sqrt(2)    inv(V_s)[0,2]*sqrt(2)    inv(V_s)[0,1]*sqrt(2)    ln(V_s)[0,0]             ln(V_s)[1,1]             ln(V_s)[2,2]             ln(V_s)[1,2]             ln(V_s)[0,2]             ln(V_s)[0,1]           
0             0.811218      1.416528e-03  1.2147778614288034e+00   6.9430960196472236e-01   7.6584443991307727e-01   6.3167126989669239e-02   1.2933003045967117e-01   -6.7849678663191706e-02  9.9985586750316435e-01   9.9992582189056489e-01   9.9983561669882814e-01   9.9853261452309073e-06   -1.3748295490876758e-05  3.0872647149967632e-05   1.4414317052969028e-04   7.4181124027670811e-05   1.6439688579071457e-04   -7.0616402794236651e-06  9.7230899134630410e-06   -2.1832675643611805e-05
1             0.807445      1.394763e-03  1.4105121075936524e+00   -1.4316282534475405e-01  -3.7249662802951083e-01  1.6011546314263277e-02   -1.1967950306433560e-01  3.4938918722138940e-01   1.0000396096410689e+00   1.0001102663613766e+00   1.0000635106130800e+00   1.9166375961745377e-05   -1.7463508670175634e-05  -3.3678434600765029e-05  -3.9608496870515006e-05  -1.1025990716100951e-04  -6.3508428310972656e-05  -1.3551349935995863e-05  1.2347767397107381e-05   2.3812381378377855e-05 
2             0.821173      2.005342e-03  3.4366064509997241e-01   8.1672100844263251e-01   -1.6460904112256015e-02  9.2964100845029634e-03   -1.1293093676934003e-01  -3.3496545923575149e-01  1.0000103502758357e+00   1.0001673830691653e+00   1.0000446355555253e+00   2.2117049188320302e-05   -2.5933387029719200e-05  3.6739917467273129e-05   -1.0349716726452111e-05  -1.6736860253952315e-04  -4.4634268989378094e-05  -1.5637695933208847e-05  1.8337372809557704e-05   -2.5976879750392947e-05
```

**Comment by psavery** [comment]
Hi @donald-e-boyce!

I just gave the GUI a try myself, and I actually got the same exact same completeness values as you did for your `hexrd` command line run. To help us test this more easily, I made a state file where all of the correct options are set. This is in the examples repository and is located [here](https://github.com/HEXRD/examples/blob/master/NIST_ruby/multiruby_dexelas/include/dexelas_id3a_20200130.h5). You still need to load in the images (I didn't write them to the state file because, since they are saved in the frame cache mode, if we write them to the HDF5 state file, the full arrays are stored and it blows up in disk space - maybe we can figure out how to write in frame cache mode to the state file also if needed).

I see there are other small discrepancies in values and I suspect that maybe the issue is [this](https://github.com/HEXRD/hexrdgui/issues/1357)? (maybe you didn't select all HKLs above the `min_sfac_ratio` in the fit grains step?). That's something I think we need to improve upon in the GUI (I mentioned this in the issue).

However, even when I only had the same 6 HKLs selected for `fit-grains` in the GUI as were used for eta omega map generation in the command line, I still get matching completeness values (only, the other values such as the `exp_map_c` have some discrepancy). If I make the HKLs match (via [this](https://github.com/HEXRD/hexrdgui/issues/1357)), then the values are nearly identical.

**Comment by donald-e-boyce** [comment]
Today, I also got the same values, exactly for completeness, but very slightly different in the other fields. 

I used your state file, loading it from the command line. Nice feature by the way. I don't know exactly what I did yesterday. Today I didn't pick any HKLs for the fit-grains.  I did notice that when I did an indexing, it came back with completeness values of one, but orientations, etc. were not computed.
